charts/rancher-turtles/templates/_helpers.tpl

```diff
@@ -1,9 +1,22 @@
 {{/*
     This removes the part after the + in the kubernetes version string.
     v1.27.4+k3s1 -> v1.27.4
+    v1.26.12-eks-5e0fdde -> v1.26.12
+    v1.26.12-gke.1 -> v1.26.12
     v1.28.0 -> v1.28.0
 */}}
 {{- define "strippedKubeVersion" -}}
-{{- $parts := split "+" .Capabilities.KubeVersion.Version -}}
-{{- print $parts._0 -}}
+    {{- if (.Capabilities.KubeVersion.Version | contains "-eks-") -}}
+        {{- $parts := split "-eks-" .Capabilities.KubeVersion.Version -}}
+        {{- print $parts._0 -}}
+    {{- else if (.Capabilities.KubeVersion.Version | contains "-gke.") -}}
+        {{- $parts := split "-gke." .Capabilities.KubeVersion.Version -}}
+        {{- print $parts._0 -}}
+    {{- else if (.Capabilities.KubeVersion.Version | contains "-aks") -}}
+        {{- $parts := split "-aks" .Capabilities.KubeVersion.Version -}}
+        {{- print $parts._0 -}}
+    {{- else -}}
+        {{- $parts := split "+" .Capabilities.KubeVersion.Version -}}
```

Comment:
{{- $parts := split "-" .Capabilities.KubeVersion.Version -}}

I think we can just strip the suffix here generically? WDYT? But just a nit, lgtm

Reply:
I did think about just stripping from `-` so went with something specific to each case.

Happy to change it.